### ISSUE TITLE
TSC generates type definition files it considers itself as containing an error.

### ISSUE THREAD

**Issue by mike-lischke** [microsoft/TypeScript#60207]
### 🔎 Search Terms

"#private;", "Private identifiers are only available when targeting ECMAScript 2015 and higher.", "ts(18028)"

### 🕗 Version & Regression Information

I have no idea when this happened the first time, I saw the error quite a few times in the past, but ignored them, as that only was shown in type definition files.

### ⏯ Playground Link

https://www.typescriptlang.org/play/?target=9&jsx=0#code/MYGwhgzhAEAqCmEAu0DeAoa0DEAzAlvCACYCCAXNAHYCuAtgEbwBO0AvNAIwDcmOBRYgCFKyZvioBzdtABESRElm8+ABxoMQ+YNEnwUAkqQAUASkq1GLNHyzN9NZlWhIAFvggA6PISO8sAL7o6AFAA

### 💻 Code

```ts
class Test {
  #fieldA: number = 1;
  #fieldB: string = "test";

  public get fieldA(): number {
    return this.#fieldA;
  }

}
```

### 🙁 Actual behavior

TSC generates a .d.ts file containing a line "#private;", which not only makes no sense (to me), but causes the mentioned error to show up when that file is opened in VS Code.

![Image](https://github.com/user-attachments/assets/252f9eb1-2879-4b7d-936f-9f0992c09fd4)

### 🙂 Expected behavior

I either expect this line not to be in the generated .d.ts file or TypeScript to accept the syntax it has written itself.

### Additional information about the issue

I asked on [Stackoverflow](https://stackoverflow.com/questions/79080821/getting-error-private-identifiers-are-only-available-when-targeting-ecmascript) before reporting here, with some more details.

**Comment by MartinJohns** [comment]
This is working as intended. The `#private` is added to make the class nominally typed (instead of structurally typed). The created declaration file is also valid when you target ECMAScript 2015 or higher.

**Comment by mike-lischke** [comment]
OK, that explains why this line is added, but why does TS complain about it then?

**Comment by MartinJohns** [comment]
Because you open the file in a context without a `tsconfig.json` file, or where the `tsconfig.json` has a target lower than ECMAScript 2015. The default is ECMAScript 5, which does not support private fields.

Note that the error does not mention invalid syntax, it says "only available when targeting ECMAScript 2015 and higher".

**Comment by mike-lischke** [comment]
Yes, that's what I thought too, but I also assumed that VS Code takes the project's tsconfig.json file into account. Am I wrong?

**Comment by mike-lischke** [comment]
I believe this issue is not a problem of TS per se, so I'm going to close it. Thanks @MartinJohns for explaining the `#private;' line.
